### ISSUE TITLE
External calls

### ISSUE THREAD

**Issue by Aurel300** [flintrocks/flint#36]
External calls into EVM contracts, likely not written in Flint.

[Updated proposal](https://github.com/flintrocks/flint/blob/master/proposals/0003-external-calls.md) is now merged into `master`.

# Tickets created

Tickets that can be worked on now:

 - #68 Parse `do ... catch` statements
 - #69 Parse `if let ...` statements 
 - #70 Parse external calls
 - #71 External traits

Future work, i.e. tickets that are blocked by one or more of the above:

 - #72 Semantic checks for external calls
 - #73 Generate IR for external calls
 - #74 Re-entrancy protection during external calls
